connectors/aws/aws-textract/src/main/java/io/camunda/connector/textract/model/TextractRequestData.java

```diff
@@ -16,60 +16,73 @@
             condition =
                 @TemplateProperty.PropertyCondition(
                     property = "input.documentLocationType",
                     equals = "S3"),
             constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
         String documentS3Bucket,
     @TemplateProperty(
-            group = "input",
-            label = "Document path",
-            description = "S3 document path to be processed",
+            group = "document",
+            label = "Document name",
+            description = "S3 document name of the document that should be analyzed.",
             condition =
                 @TemplateProperty.PropertyCondition(
                     property = "input.documentLocationType",
                     equals = "S3"),
             constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
         String documentName,
     @TemplateProperty(
-            group = "input",
+            group = "document",
             label = "Document version",
-            description = "S3 document version to be processed",
+            description = "S3 document version of the document that should be analyzed.",
             optional = true,
             condition =
                 @TemplateProperty.PropertyCondition(
                     property = "input.documentLocationType",
                     equals = "S3"))
         String documentVersion,
+    @TemplateProperty(
+            group = "document",
+            label = "Camunda Document",
+            description = "The Camunda document of the process that should be analyzed.",
+            feel = Property.FeelMode.required,
+            type = TemplateProperty.PropertyType.String,
+            condition =
+                @TemplateProperty.PropertyCondition(
+                    property = "input.documentLocationType",
+                    equals = "UPLOADED"),
+            constraints = @TemplateProperty.PropertyConstraints(notEmpty = true))
+        Document document,
+    @TemplateProperty(
+            label = "Execution type",
+            group = "input",
+            type = TemplateProperty.PropertyType.Dropdown,
+            defaultValue = "POLLING",
+            feel = FeelMode.disabled,
+            description =
+                "How the document should be processes. See more info in the <a href=\"https://docs.camunda.io/docs/components/connectors/out-of-the-box-connectors/amazon-textract/#execution-types\" target=\"_blank\">documentation</a>.",
+            condition =
+                @TemplateProperty.PropertyCondition(
+                    property = "input.documentLocationType",
+                    equals = "S3"))
+        @NotNull
+        TextractExecutionType executionType,
     @TemplateProperty(
             label = "Analyze tables",
+            description =
+                "Select this to return information about the tables that are detected in the input document.",
```

Comment:
[OCD mode on] Some of the descriptions in this file end with dots, some of them don't :)